bench2.c

```
@@ -0,0 +1,178 @@
+    size_t len = strnlen(str, NAME_MAX);
+    for (size_t i = 0; i < len; i++) {
+        c = str[i];
+        if (i == len - 1 && c == '/') continue; // Ignore trailing slash
+        hash = ((hash << 5) + hash) + c; /* hash * 33 + c */
+    }
+    return hash;
+}
+
+LinkHashSet *LinkHashSet_new(int capacity) {
+    LinkHashSet *set = calloc(1, sizeof(LinkHashSet));
+    set->capacity = capacity;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)